src/Microsoft.ML.Data/Utilities/TypeParsingUtils.cs

```diff
@@ -57,23 +57,21 @@ public static KeyType ConstructKeyType(DataKind? type, KeyRange range)
 
             DataKind kind;
             KeyType keyType;
-            kind = type ?? DataKind.U4;
+            kind = type ?? DataKind.U8;
             Contracts.CheckUserArg(KeyType.IsValidDataKind(kind), nameof(TextLoader.Column.Type), "Bad item type for Key");
-            Contracts.CheckUserArg(range.Min >= 0, nameof(range.Min), "min must be non-negative");
```

Comment:
>Min [](start = 41, length = 3)

Just thought of this... what is happening to `KeyRange.Min`? We should probably be asserting that `Min` is 0, if we are to keep it at all. (Of course, in the absence of support for min values we could just change the format rather entirely so that what is currently `U4[0-100]` could be parsed as `U4[100]`, but maybe we want to hold off on that for now.)

Reply:
I decided to remove the Min from KeyRange as KeyType and KeyRange should map to each other. I answered in another of your comments that I agree we should match the string format of KeyRange not to have the Min and to have a determined Count.

---
In reply to: [249262433](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249262433) [](ancestors = 249262433)